### DISCUSSION TITLE
3.2rc0

### DISCUSSION THREAD

**Discussion by snowman2** [pyproj4/pyproj#925] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/pyproj4/pyproj/releases/tag/3.2rc0'>3.2rc0</a>.</em>

**Comment by snowman2** [comment]
The plan is to release with PROJ 8.1.1 in the wheels.

Release Candidate 0 Status:
- [x] Linux/OSX wheels (https://github.com/pyproj4/pyproj-wheels/pull/36)
- [x] Windows wheels (@cgohlke)
- [x] Conda Forge (https://github.com/conda-forge/pyproj-feedstock/pull/101)
- [x] Debian (@sebastic)
- [x] Fedora (@jdkloe)

Plan to release sometime this weekend. Testing is always appreciated, but definitely not required.

**Comment by sebastic** [comment]
Debian package builds (with PROJ 7.2.1) looking good so far:

 https://buildd.debian.org/status/package.php?p=python-pyproj&suite=experimental

**Reply by snowman2** [reply]
Great, thanks for testing 👍

**Comment by jdkloe** [comment]
The3.2rc0 version compiles fine with Fedora Rawhide, but I get 11 test failures, all related to  test/test_cli.py. I guess this could be another python3.10 issue. They all end with a "Permission denied" error:

============================================================================== short test summary info ==============================================================================
FAILED test/test_cli.py::test_main[input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_main__verbose[-v-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_main__verbose[--verbose-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync[option0-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync[option1-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--list-files0-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--source-id-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--area-of-use-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--bbox-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--list-files1-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
FAILED test/test_cli.py::test_sync__all__exclusive_error[--file-input_command0] - PermissionError: [Errno 13] Permission denied: 'pyproj'
======================================================= 11 failed, 774 passed, 4 skipped, 24 deselected, 9 warnings in 17.35s =======================================================

I am trying to figure out now what is going on.

**Reply by jdkloe** [reply]
Problem seems to be related to the fact that I need to run the tests with the  -m "not network" option. It looks like there are a number of tests in test_cli.py that do try to access or open a networkconnection, even if this not network option is used during a test run.
Adding "@pytest.mark.network" in 4 places in test/test_cli.py makes the testrun pass again.

I added "@pytest.mark.network" at:
line 22 just bedore def test_main
line 34 just before test_main_verbose
line 52 just before def test_sync
line 199 just before def test_sync_all_exclusive_error

If you think this is right I will be happy to provide a pull request with the change.

**Reply by snowman2** [reply]
Thanks for testing 👍. For debian, they skip the cli tests with `-m "not cli"` as well. The tests you listed that are failing shouldn't need the network.

**Reply by snowman2** [reply]
Related #740, #741

**Reply by jdkloe** [reply]
The -m "not cli" option does not solve it on my side. In stead the number of fails grows to 14. Or am I using the wrong syntax here?
I executed: pytest-3  -m "not network"  -m "not cli"
and get:
============================================================================== short test summary info ==============================================================================
FAILED test/test_cli.py::test_sync_download - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_cli.py::test_sync_download__directory - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_cli.py::test_sync_download__system_directory - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_cli.py::test_sync__download_grids - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_doctest_wrapper.py::test_doctests__network - AssertionError: 4 of the doctests failed
FAILED test/test_sync.py::test_get_transform_grid_list - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__bbox__antimeridian - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__bbox__out_of_bounds - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__source_id - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__contains - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__file - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test_get_transform_grid_list__area_of_use - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_sync.py::test__load_grid_geojson_old_file - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
FAILED test/test_transformer.py::test_transformer_group__download_grids - urllib.error.URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
====================================================== 14 failed, 763 passed, 4 skipped, 32 deselected, 10 warnings in 20.22s =======================================================

**Comment by jdkloe** [comment]
Found it, I should have used:
pytest-3  -m "not network and not cli" 
then all selected tests pass.
So there is no showstopper for this release on my side. (but still I don't understand why these cli tests fail).

**Reply by snowman2** [reply]
Are you able to execute `pyproj -v` after installing pyproj?

**Reply by snowman2** [reply]
It is added during setup: [ref](https://github.com/pyproj4/pyproj/blob/3812a2b693aacb62387199dc9878ad0189bad103/setup.cfg#L42-L44)

**Reply by jdkloe** [reply]
yes I am, after I added /usr/local/bin to the PATH setting (it is not by default part of the PATH setting in the mock environment)

**Reply by jdkloe** [reply]
This is the output I get from pyproj -v:

pyproj info:
    pyproj: 3.2.0rc0
      PROJ: 8.1.1
  data dir: /usr/share/proj
user_data_dir: /builddir/.local/share/proj

System:
    python: 3.10.0rc1 (default, Aug  3 2021, 00:00:00) [GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
executable: /usr/bin/python3
   machine: Linux-5.13.12-200.fc34.x86_64-x86_64-with-glibc2.34.9000

Python deps:
   certifi: 2020.12.05
       pip: None
setuptools: 57.4.0
    Cython: 0.29.24

**Reply by snowman2** [reply]
Strange that you can run it in the console, but the test isn't allowed to execute it.

**Reply by jdkloe** [reply]
Turns out I made a mistake on my side. I modified the PATH to include the build dir after doing only the build step but not the install step. But forgot to undo that change and have it point to the /usr/local/bin dir as well after installing. If I add that dir to the PATH everything runs smooth and the "not cli" setting is not needed at all.
Seems I was confused by the errors printed by pytest which are not very clear at all if the pyprint script cannot be found on the search path.
Sorry for causing this confusion.

**Reply by snowman2** [reply]
Glad you were able to find a resolution & thanks again for testing :+1:

**Comment by snowman2** [comment]
https://github.com/pyproj4/pyproj/releases/tag/3.2.0
Will be on pypi soon.

**Comment by cgohlke** [comment]
I uploaded Windows builds at https://www.lfd.uci.edu/~gohlke/pythonlibs/#pyproj

They are now linked to proj_8_1.dll. Static linking was no longer feasible. All tests pass, even on CPython 3.10 and PyPy3.7

**Reply by snowman2** [reply]
Thanks @cgohlke :+1:
